controller/dataplane/owned_deployment.go

```diff
@@ -271,6 +280,43 @@ func listOrReduceDataPlaneDeployments(
 	return false, &deployments[0], nil
 }
 
+func podTemplateSpecHasRestartAnnotation(template *corev1.PodTemplateSpec) (string, bool) {
+	if template == nil || template.Annotations == nil {
+		return "", false
+	}
+	return template.Annotations[restartAnnotationKey], template.Annotations[restartAnnotationKey] != ""
+}
+
+// isRecentDeploymentRestart detects if a deployment is undergoing a recent restart operation.
+// It checks for the presence of the kubectl restart annotation and verifies if it's recent (within 5 minutes).
+// Returns true if a recent restart is detected, false otherwise.
+func isRecentDeploymentRestart(template *corev1.PodTemplateSpec, logger logr.Logger) bool {
+	// Check if we have a fresh restart annotation (from the last 5 minutes)
+	// This prevents us from treating all deployments with old restart annotations
+	// as perpetually in a restart state, fixing issue #1390
+	restartTimeStr, hasRestartAnnotation := podTemplateSpecHasRestartAnnotation(template)
+	if hasRestartAnnotation {
+		// Only treat it as a restart if the timestamp is recent (within 5 minutes)
+		restartTime, err := time.Parse(time.RFC3339, restartTimeStr)
+		if err == nil {
+			// Check if restart annotation is less than 5 minutes old
+			// We use 5m here as threshold as that correlates with constants used by Kubernetes itself.
+			// Restart annotations older than this duration shouldn't cause a reconciliation effect.
+			// See: https://github.com/kubernetes/kubernetes/blob/82db38a23c7820b1924d89f458fd368023f3980c/pkg/controller/namespace/config/v1alpha1/defaults.go#L41
+			if time.Since(restartTime) < 5*time.Minute {
+				log.Debug(logger, "detected recent restart operation", "isRestartOperation", true, "restartTime", restartTime)
+				return true
+			} else {
+				log.Debug(logger, "found old restart annotation, not treating as restart", "restartTime", restartTime)
+			}
+		} else {
+			log.Debug(logger, "detected restart with unparseable timestamp", "timestamp", restartTimeStr)
+			return true // If we can't parse time, assume it's a restart for safety
+		}
+	}
+	return false
```

Comment:
Can we make the happy path the least indented?

So e.g. 

```
	if !hasRestartAnnotation {
		return false
	}
```

```
	restartTime, err := time.Parse(time.RFC3339, restartTimeStr)
	if err != nil {
		log.Debug(logger, "detected restart with unparseable timestamp", "timestamp", restartTimeStr)
		return true // If we can't parse time, assume it's a restart for safety
	}
	// NOTE: No else clause needed here
	...
```

related: https://maelvls.dev/go-happy-line-of-sight/